src/crh/constraints.rs

```diff
@@ -23,3 +23,22 @@ pub trait FixedLengthCRHGadget<H: FixedLengthCRH, ConstraintF: Field>: Sized {
         input: &[UInt8<ConstraintF>],
     ) -> Result<Self::OutputVar, SynthesisError>;
 }
+
+pub trait TwoToOneCRHGadget<H: TwoToOneCRH, ConstraintF: Field>: Sized {
+    type OutputVar: EqGadget<ConstraintF>
+        + ToBytesGadget<ConstraintF>
+        + CondSelectGadget<ConstraintF>
+        + AllocVar<H::Output, ConstraintF>
+        + R1CSVar<ConstraintF>
+        + Debug
+        + Clone
+        + Sized;
+
+    type ParametersVar: AllocVar<H::Parameters, ConstraintF> + Clone;
```

Comment:
What goes in here? Is this just for full generality in case someone wanted to construct a hashes parameters in circuit?

Reply:
(Also we can handle this in a follow-on PR though)

Reply:
Yes, this is in case when someone wanted to construct hashes params in circuit (not sure if such circumstance exists).

Reply:
@npwardberkeley Hi Nicholas, I just checked git history and you seem to be the original author for this part. Quick question, is there any reason to use `ParametersVar` instead of just native version of `Parameters`?

Reply:
Sorry, I don't think I actually wrote this, just moved it; so I'm not sure

Reply:
oops... ; sorry for the spam

Reply:
@ValarDragon Adding `ParametersVar` do require more constraints, so we can maybe allow an option to use native parameters instead. (I believe this will need quite a bit of refactor on existing implementations of the trait, so I suggest doing this in a separate PR)

Reply:
That makes sense, lets file an issue for discussing it, and potentially handle it in a follow-on pr

Reply:
just filed an issue for this